### PR TITLE
[plugin-mobile-app] Add ability to set vertical x position for swipe

### LINKED ISSUE #1774: Swipe to element step isn't working for x=1 coordinate
The step 
"When I swipe direction to element located `locator` with duration swipeDuration"
doesn't work for mobile test.

Steps:
execute steps
Given I start mobile application
When I swipe direction to element located `locator` with duration swipeDuration

Expected: swipe is executed
Actual: Not executed (element by locator is not found)
image

NOTE:
It was found with debugger that step swipes by next coordinates:
SwipeCoordinates(1,1093,1,273)
After changed manually to
SwipeCoordinates(22,1093,22,273) the test worked correctly

### PR DESCRIPTION
**Pull request by uarlouski**
Closes #1774

### CODE REVIEW

File: vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/model/SwipeDirection.java [hunk 1]
```diff
@@ -16,31 +16,41 @@
...
     private static final int VERTICAL_INDENT_COEFFICIENT = 5;
-    private static final int SWIPE_X = 1;
 
-    public abstract SwipeCoordinates calculateCoordinates(Dimension dimension);
+    public abstract SwipeCoordinates calculateCoordinates(Dimension dimension, int xOffsetPercentage);
+
+    @SuppressWarnings("MagicNumber")
+    public static int calculateX(int width, int percentage)
+    {
+        Validate.isTrue(percentage >= 0 && percentage <= 100,
```
Comment (ikalinin1) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/model/SwipeDirection.java:52]: please consider this validation at initialization level

File: vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/model/SwipeDirection.java [hunk 2]
```diff
@@ -16,31 +16,41 @@
...
+                "The x percentage value must be between 0 and 100, but got: %d", percentage);
+        return percentage == 0 ? 1 : (width - 1) * percentage / 100;
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/model/SwipeDirection.java:54]: if percentage = 0, then result = 1;
if percentage = 100, then result = width - 1;
else result = width * percentage / 100

File: vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/TouchActions.java [hunk 3]
```diff
@@ -164,7 +165,9 @@ public void swipeUntil(SwipeDirection direction, Duration swipeDuration, Boolean
      */
     public void performVerticalSwipe(int startY, int endY, Duration swipeDuration)
     {
-        swipe(new SwipeCoordinates(1, startY, 1, endY), swipeDuration);
+        int x = SwipeDirection.calculateX(genericWebDriverManager.getSize().getWidth(),
+                mobileApplicationConfiguration.getSwipeVerticalXPosition());
+        swipe(new SwipeCoordinates(x, startY, x, endY), swipeDuration);
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/TouchActions.java:170]: consider refactoring to avoid duplication

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/model/SwipeDirection.java [hunk 4]
```diff
@@ -23,24 +23,43 @@
...
+    }
+
+    @SuppressWarnings("MagicNumber")
+    private static int calculateX(int width, int percentage)
+    {
+        if (percentage == 0)
+        {
+            return 1;
+        }
+        else if (percentage == 100)
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/model/SwipeDirection.java:59]:
```suggestion
        if (percentage == 100)
```

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by ikalinin1 — APPROVED** [review]

**Review by valfirst — APPROVED** [review]

**Pull request merged**